xbatcher/loaders/torch.py

```diff
@@ -54,8 +54,8 @@ def __getitem__(self, idx) -> Tuple[Any, Any]:
 
         # TODO: figure out the dataset -> array workflow
         # currently hardcoding a variable name
-        X_batch = self.X_generator[idx]['x'].torch.to_tensor()
-        y_batch = self.y_generator[idx]['y'].torch.to_tensor()
+        X_batch = self.X_generator[idx].torch.to_tensor()
+        y_batch = self.y_generator[idx].torch.to_tensor()
```

Comment:
This works because the unit tests in `test_torch_loaders.py` are actually testing `xarray.DataArray` inputs only, and not `xarray.Dataset`. Ideally there would be unit tests for both `xr.DataArray` and `xr.Dataset` inputs, but this might expand the scope of the Pull Request a bit too much :sweat_smile:

Reply:
Before this PR, the tests used `xarray.DataArray` inputs to the batch generator, but `xarray.Dataset` inputs to the dataloaders since the batches were coerced into datasets. So I think this would be an additional breaking change in expecting `xarray.DataArray` inputs.

For the unit tests, I opened https://github.com/pangeo-data/xbatcher/issues/83 to keep track of improvements for subsequent PRs.

Reply:
What are your thoughts on backwards compatibility here @weiji14? My impression is that the hardcoded `xarray.Dataset` variable names severely restricts the utility of the data loader. So, I think this is a worthwhile change since we'd be forced to break backwards compatibility anyways eventually for flexible variable names and that working from an `xarray.DataArray` implementation is better. But we could add an `if; else` block for dataset vs. dataarray if it's necessary to maintain the past behavior.

Reply:
Yes, I agree that we should try to be backward compatible and support both `xr.DataArray` and `xr.Dataset` inputs. If you want, I can either:

1. Add the if-then block, but write the comprehensive unit tests covering `xr.Dataset`/`xr.DataArray` cases in a separate PR
2. Do the if-then block and unit-tests in a follow up PR, in order to keep this PR small.

Reply:
+1 for option 2